packages/ember-debug/lib/deprecate.js

```diff
@@ -95,8 +95,9 @@ export let missingOptionsUntilDeprecation = 'When calling `Ember.deprecate` you
 
 /**
   Display a deprecation warning with the provided message and a stack trace
-  (Chrome and Firefox only). Ember build tools will remove any calls to
-  `Ember.deprecate()` when doing a production build.
+  (Chrome and Firefox only). Ember build tools will remove any calls
+  to `Ember.deprecate()` when doing an Ember.js framework production build
+  and will make the assertion a no-op for an application production build.
```

Comment:
I cannot speak for @webark, but I think this was to eventually match the behavior to be implemented by https://github.com/ember-cli/rfcs/pull/50.

We recently came across this when we removed all custom deprecation handlers and the default Ember console.log implementation started to flood our prod consoles.

I took from my slack conversation that this RFC was to land quickly.  Which might be the motivation not to updated the docs to match current behavior only to change them back after impl?